spec/ddtrace/contrib/rest_client/request_patch_spec.rb

```diff
@@ -0,0 +1,170 @@
+require 'spec_helper'
+require 'ddtrace'
+require 'ddtrace/contrib/rest_client/request_patch'
+require 'rest_client'
+
+RSpec.describe Datadog::Contrib::RestClient::RequestPatch do
+  let(:tracer) { Datadog::Tracer.new(writer: FauxWriter.new) }
+  let(:rest_client_options) { {} }
+
+  before do
+    Datadog.configure do |c|
+      c.use :rest_client, rest_client_options.merge(tracer: tracer)
+    end
+
+    WebMock.disable_net_connect!
+    WebMock.enable!
+  end
+
+  describe 'instrumented request' do
+    let(:path) { '/sample/path' }
+    let(:host) { 'example.com' }
+    let(:url) { "http://#{host}#{path}" }
+    let(:status) { 200 }
+    let(:response) { 'response' }
+
+    subject(:request) { RestClient.get(url) }
+
+    before do
+      stub_request(:get, url)
+        .to_return(status: status, body: response)
+    end
+
+    shared_examples_for 'instrumented request' do
+      it 'creates a span' do
+        expect { request }.to change { tracer.writer.spans.first }.to be_instance_of(Datadog::Span)
+      end
+
+      it 'returns response' do
+        expect(request.body).to eq(response)
+      end
+
+      describe 'created span' do
+        subject(:span) { tracer.writer.spans.first }
+
+        context 'response is successfull' do
+          before do
```

Comment:
Minor non-blocking, but short, single line blocks like this could be reduced to the more compact `before { request }` form.